src/griffe/finder.py

```diff
@@ -137,13 +139,17 @@ def find_module(self, module_name: str) -> Path | list[Path]:  # noqa: WPS231
                     abs_path = path / choice
                     if abs_path in path_contents:
                         if abs_path.suffix:
-                            return abs_path
+                            found_paths.append(abs_path)
```

Comment:
👀 : this of course comes at the loss of early exiting.  So could perhaps slow things down with large search paths.  If you'd prefer that we only look for a pyi file once the original `py` file has been found, let me know

Reply:
what about `.pyi` files that are there for non `.py` files, like `.py[cod]`, and `.so` files which we're currently skipping?